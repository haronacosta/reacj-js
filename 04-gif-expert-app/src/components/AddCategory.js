import React, { useState } from 'react';

const AddCategory = () => {
  const [inputValue, setInputValue] = useState('Hola Mundo');

  const handleInputChange = (e) => {
    const newInputValue = e.target.value;

    setInputValue(newInputValue);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    console.log(e.target);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        name="newCategory"
        id="newCategory"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

export default AddCategory;
