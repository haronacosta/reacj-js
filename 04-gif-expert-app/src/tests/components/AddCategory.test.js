import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en el componente AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Se debe de mostrar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Verificamos el funcionamiento del input, debe de cambiar la caja de texto', () => {
    const value = 'Hola mundo';

    wrapper.find('input').simulate('change', { target: { value } });

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('No debe de ser llamada la funciÃ³n setCategories con un input en blanco', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });
});
