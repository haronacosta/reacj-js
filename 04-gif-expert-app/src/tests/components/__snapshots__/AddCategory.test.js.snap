// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory Se debe de mostrar correctamente el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    id="newCategory"
    name="newCategory"
    onChange={[Function]}
    placeholder="Ingrese un valor"
    type="text"
    value=""
  />
</form>
`;
